posts/capture-data-history-with-scd2-using-databricks-delta-live-tables.md

```diff
@@ -0,0 +1,84 @@
+---
+authors:
+- Theo Lebrun
+tags:
+- Data
+- Databricks
+- Apache Spark
+- Python
+- ETL
+date: 2023-01-23T14:50:55.000Z
+title: "Capture Data History With SCD2 Using Databricks Delta Live Tables"
+image: 
+---
+
+[Delta Live Tables](https://www.databricks.com/product/delta-live-tables) is a great way to build and manage reliable batch and streaming data pipelines on your Databricks Lakehouse. Your ETL pipelines will be simplier thanks to multiple out of the box features while having access to useful functions from the DLT module. For more information, I recommend reading this [Quickstart article](https://docs.databricks.com/workflows/delta-live-tables/delta-live-tables-quickstart.html) that will get you started with DLT.
+
+# How Databricks Delta Live Tables (DLT) simplify your life
+
+Delta Live Tables is a framework for building reliable, maintainable, and testable data processing pipelines. You define the transformations to perform on your data, and Delta Live Tables manages task orchestration, cluster management, monitoring, data quality, and error handling.
+
+One of the great things about DLT is that you can use Python decorators to handle things like table dependencies or data quality. Coupled with [Auto Loader](https://docs.databricks.com/ingestion/auto-loader/index.html), you will be able to load and transform your data with a very low amount of code. For more details and real examples, I suggest to take a look at these [example notebooks](https://github.com/databricks/delta-live-tables-notebooks).
+
+You can also build a streaming pipeline that runs all the time and will ingest your streaming data. Migrating from a regular batch pipeline requires minimal changes as all the syntax is the same and you will able to perform complex operations like joins and aggregations.
+
+# What is SCD2 (Slowly Changing Dimensions Type 2)?
+
+Storing daily data in your data lake is the traditional way of operating and will provide historical data. Nothing wrong by doing that, and in most cases, having daily data is enough for your internal processes. But what if you could enhance your data by being able to know its period of validity?
+
+For example, let's say that your system is saving estimate data for stocks and those estimates change over time. Portfolio managers are gonna be able to enhance their decision making by easily knowing when a given estimate was valid. Here is how it would look like in the tables:
```

Comment:
Portfolio managers are what? are **going to want to** enhance their decision making